sysl2/sysl/integration/deps.go

```diff
@@ -0,0 +1,304 @@
+package integration
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/anz-bank/sysl/src/proto"
+	"github.com/anz-bank/sysl/sysl2/sysl/utils"
+	"github.com/yale8848/stream"
+)
+
+type AppElement struct {
+	Name     string
+	Endpoint string
+}
+
+type AppDependency struct {
+	Self   *AppElement
+	Target *AppElement
+}
+
+type DependencySet struct {
+	Deps   map[*AppDependency]struct{}
```

Comment:
How is this app dependency set?
you are working with pointer to a structs, but that struct could have same values and still be in the set.

Is there a test that proves `DependencySet` is actually a set?

Reply:
Add a new test file for validating the DependencySet. Please check integration_dependency_set_test.sysl. When hit if/else branch, we may get duplicate target application and target endpoint, so we'd use set structure to filter the duplicate record.